adapter/ipListChecker/ipListChecker.go

```diff
@@ -137,87 +164,63 @@ func (l *listChecker) CheckList(symbol string) (bool, error) {
 }
 
 // Typed accessors for the atomic list
-func (l *listChecker) getList() listState {
+func (l *listChecker) getListState() listState {
 	return l.atomicList.Load().(listState)
 }
 
 // Typed accessor for the atomic list
-func (l *listChecker) setList(list listState) {
-	l.atomicList.Store(list)
-}
-
-// Updates the list by polling from the provider on a fixed interval
-func (l *listChecker) listRefresher() {
-	refreshTicker := time.NewTicker(l.refreshInterval)
-	purgeTimer := time.NewTimer(l.ttl)
-
-	defer refreshTicker.Stop()
-	defer purgeTimer.Stop()
-
-	for {
-		select {
-		case <-refreshTicker.C:
-			// fetch a new list and reset the TTL timer
-			l.refreshList()
-			purgeTimer.Reset(l.ttl)
-
-		case <-purgeTimer.C:
-			// times up, nuke the list and start returning errors
-			list := l.getList()
-			list.payload = nil
-			list.payloadSha = [20]byte{}
-			l.setList(list)
-
-		case <-l.closing:
-			return
-		}
-	}
+func (l *listChecker) setListState(ls listState) {
+	l.atomicList.Store(ls)
 }
 
 // represents the format of the data in a list
 type listPayload struct {
 	WhiteList []string `yaml:"whitelist" required:"true"`
 }
 
-func (l *listChecker) refreshList() {
+func (l *listChecker) fetchList() error {
 	l.log.Infof("Fetching list from %s", l.providerURL)
 
-	list := l.getList()
-
 	resp, err := l.client.Get(l.providerURL)
 	if err != nil {
-		list.fetchError = err
-		l.setList(list)
+		ls := l.getListState()
+		ls.fetchError = err
+		l.setListState(ls)
 		l.log.Warningf("Could not connect to %s: %v", l.providerURL, err)
```

Comment:
(here and elsewhere) i think we should try to avoid logging an error and returning it as well.  if we want to log the error, we should do it somewhere where it is actually handled (rather than passed up).